tests/conftest.py

```diff
@@ -0,0 +1,238 @@
+    """Instantiate and return Indexer client object."""
+    return indexer.IndexerClient(indexer_token, indexer_address)
+
+
+def _kmd_client(kmd_address="http://localhost:4002", kmd_token="a" * 64):
+    """Instantiate and return a KMD client object."""
+    return kmd.KMDClient(kmd_token, kmd_address)
+
+
+# Env helpers
+
+
+@dataclass
+class Account:
+    address: str
+    private_key: Optional[str]  # Must be explicitly set to None when setting `lsig`.
+    lsig: Optional[transaction.LogicSig] = None
+
+    def __post_init__(self):
+        assert self.private_key or self.lsig
+
+    def mnemonic(self) -> str:
+        return mnemonic.from_private_key(self.private_key)
+
+    def is_lsig(self) -> bool:
+        return bool(not self.private_key and self.lsig)
+
+    @classmethod
+    def create(cls) -> "Account":
+        private_key, address = account.generate_account()
+        return cls(private_key=private_key, address=str(address))
+
+    @property
+    def decoded_address(self):
+        return encoding.decode_address(self.address)
+
+
+def get_kmd_accounts(
+    kmd_wallet_name="unencrypted-default-wallet", kmd_wallet_password=""
+):
+    kmd_client = _kmd_client()
+    wallets = kmd_client.list_wallets()
+
+    walletID = None
+    for wallet in wallets:
+        if wallet["name"] == kmd_wallet_name:
+            walletID = wallet["id"]
+            break
+
+    if walletID is None:
+        raise Exception("Wallet not found: {}".format(kmd_wallet_name))
+
+    walletHandle = kmd_client.init_wallet_handle(walletID, kmd_wallet_password)
+
+    try:
+        addresses = kmd_client.list_keys(walletHandle)
+
+        privateKeys = [
+            kmd_client.export_key(walletHandle, kmd_wallet_password, addr)
+            for addr in addresses
+        ]
+
+        kmdAccounts = [
+            Account(address=addresses[i], private_key=privateKeys[i])
+            for i in range(len(privateKeys))
+        ]
+    finally:
+        kmd_client.release_wallet_handle(walletHandle)
+
+    return kmdAccounts
+
+
+## Teal Helpers
+
+
+def logged_bytes(b: str):
+    return bytes(b, "ascii").hex()
+
+
+def logged_int(i: int):
+    return i.to_bytes(8, "big").hex()
+
+
+def assert_stateful_output(expr: Expr, output: List[str]):
+    assert expr is not None
+
+    src = compile_stateful_app(expr)
+    assert len(src) > 0
+
+    compiled = fully_compile(src)
+    assert len(compiled["hash"]) == 58
+
+    app_id = create_app(
+        compiled["result"],
+        transaction.StateSchema(0, 16),
+        transaction.StateSchema(0, 64),
+    )
+
+    result = call_app(app_id)
+
+    assert result == output
+
+
+def assert_output(expr: Expr, output: List[str], **kwargs):
+    assert expr is not None
+
+    src = compile_app(expr)
+    assert len(src) > 0
+
+    compiled = fully_compile(src)
+    assert len(compiled["hash"]) == 58
+
+    result = execute_app(compiled["result"], **kwargs)
+    assert result == output
+
+
+def compile_app(method: Expr):
```

Comment:
```suggestion
def compile_app(method: Expr, version: int = 5):
```
nice to expose the version